### DISCUSSION TITLE
Share read only calendar: only the users calendars are shown

### DISCUSSION THREAD

**Discussion by r00tr4v3n** [Kozea/Radicale#1612] [Q&A server] [answered]
I'm trying to setup two users (and more) with shared work/projects calendars and a private one.
When accessing the work calendar from user2 with user1, I get  all calendars from user 1 again. Thus, I get the same calendars from user1 twice. I can download the work calendar from user2 via browser.
This URL gets me the calendars from user1 instead of 2 when connected with user1 credentials: https://domain.com/user2/work/ 
Any idea why?

```
[root]
user: .+
collection:
permissions: rR

[share_alice_work]
user: bob
collection: alice/work
permissions: r


# Allow reading and writing principal collection (same as username)
[principal]
user: .+
collection: {user}
permissions: RW

# Allow reading and writing calendars and address books that are direct
# children of the principal collection
[calendars]
user: .+
#user: {user}
collection: {user}/[^/]+
permissions: rw

[share_family]
user: alice
collection: bob/family
permissions: rw


[share_robot]
user: robot
collection: bob/projects
permissions: rw

```

**Comment by pbiering** [comment]
Can you paste the related directory structure of the collections and their symlinks like shown in Wiki
https://github.com/Kozea/Radicale/wiki/Sharing-Collections

**Reply by r00tr4v3n** [reply]
I gave up and used the 'owner_write' setting with symlinks but this allows user one to write in user2_work.
```
.
└── collections
    └── collection-root
        ├── admin
        │   ├── a330c1b4-632d-c9cd-d3fd-1a3a50704784
        ├── user2
        │   ├── work
        │   ├── share -> ../user1/share/
        │   ├── private
        │   └── projects -> ../user1/projects/
        ├── user1
        │   ├── share
        │   ├── user2_work -> ../user2/work/
        │   ├── private
        │   └── projects -> ../bot/projects/
        └── bot
            └── projects
```

**Comment by pbiering** [comment, marked as answer]
the rights file is parsed sequentially and first match wins, can you try add following **before** generic section starts?

```
[calendars-user2-share]
user: user2
collection: {user}/share
permissions: r

[calendars-user2-projects]
user: user2
collection: {user}/projects
permissions: r

[calendars-user1-share]
user: user1
collection: {user}/user2_work
permissions: r

[calendars-user1-projects]
user: user1
collection: {user}/projects
permissions: r
```

You can use option `--logging-rights-rule-doesnt-match-on-debug` to get information about rights rule match process.
